python/lib/modeldata/dmod/modeldata/data/object_store_dataset.py

```diff
@@ -166,12 +183,20 @@ def _decode_object_name_to_file_path(self, object_name: str) -> str:
         """
         return "/".join(object_name.split(self._OBJECT_NAME_SEPARATOR))
 
+    def _gen_dataset_serial_obj_name(self, dataset_name: str) -> str:
+        return self._SERIALIZED_OBJ_NAME_TEMPLATE.format(dataset_name)
+
     # TODO: add stuff for threading
-    def _push_file(self, bucket_name: str, file: Path, bucket_root: Path, do_checks: bool = True) -> ObjectWriteResult:
+    def _push_file(self, bucket_name: str, file: Path, bucket_root: Optional[Path] = None, dest: Optional[str] = None,
```

Comment:
can we do this `async`?

Reply:
Possibly, but I'd like to kick that can down the road.  We may be able to take advantage of threads here more directly, since it has to do with IO.